src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1148,6 +1149,61 @@ impl ToQemuParams for Rtc {
     }
 }
 
+// RngDevice represents a random number generator device.
+#[derive(Debug)]
+struct RngDevice {
+    // id is the device ID
+    id: String,
+
+    // filename is the entropy source on the host
+    filename: String,
+
+    // max_bytes is the bytes allowed to guest to get from the host’s entropy per period
+    max_bytes: u32,
+
+    // period is duration of a read period in seconds
+    period: u32,
+
+    // transport is the virtio transport for this device.
+    transport: String,
+}
+
+impl RngDevice {
+    fn new() -> RngDevice {
+        RngDevice {
+            id: "rng0".to_owned(),
+            filename: "/dev/urandom".to_owned(),
+            max_bytes: 1024,
+            period: 1000,
+            transport: "virtio-rng-pci".to_owned(),
+        }
+    }
+}
+
+#[async_trait]
+impl ToQemuParams for RngDevice {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        let mut object_params = Vec::new();
+        let mut device_params = Vec::new();
+
+        object_params.push("rng-random".to_owned());
+        object_params.push(format!("id={}", self.id));
+        object_params.push(format!("filename={} ", self.filename));
+
+        device_params.push(format!("-device {}", self.transport));
```

Comment:
isn't `-device` here duplicated as is already added at line 1201? So maybe:

```
device_params.push(self.transport.clone());
```

Reply:
Yes, my bad

Reply:
Rust is snake_case, so no